<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot name="rob">
<gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      <robotNamespace>/rob</robotNamespace>
    </plugin>
    </gazebo>
  <link
    name="main_body">
    <inertial>
      <origin
        xyz="0.0599999521743748 -0.149999970206552 5.68827410385838E-09"
        rpy="0 0 0" />
      <mass
        value="4.1493" />
      <inertia
        ixx="0.0313"
        ixy="-6.6806E-09"
        ixz="1.4856E-08"
        iyy="0.0085"
        iyz="-1.5E-08"
        izz="0.0391" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/main_body.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/main_body.STL" />
      </geometry>
    </collision>
  </link>
  <gazebo reference="main_body">
    <kp>100000.0</kp>
    <kd>100000.0</kd>
    <mu1>1000000.0</mu1>
    <mu2>1000000.0</mu2>
    <material>Gazebo/Grey</material>
  </gazebo>
  <link
    name="upper_front_left">
    <inertial>
      <origin
        xyz="-2.36636892858577E-08 0.0244785713376747 -0.0318067000775048"
        rpy="0 0 0" />
      <mass
        value="0.16224" />
      <inertia
        ixx="0.00015"
        ixy="2.269E-10"
        ixz="1.2008E-10"
        iyy="3.27E-05"
        iyz="2.459E-05"
        izz="0.000133" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/upper_front_left.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/upper_front_left.STL" />
      </geometry>
    </collision>
  </link>
  <gazebo reference="upper_front_left">
    <kp>100000.0</kp>
    <kd>100000.0</kd>
    <mu1>1000000.0</mu1>
    <mu2>1000000.0</mu2>
    <material>Gazebo/Grey</material>
  </gazebo>
  <joint
    name="shoulder_left_front"
    type="revolute">
    <origin
      xyz="0.141 -0.225 0"
      rpy="-1.5708 -0.52136 1.5708" />
    <parent
      link="main_body" />
    <child
      link="upper_front_left" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-15"
      upper="15"
      effort="10"
      velocity="1.0" />
  </joint>
    <transmission name="tran1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="shoulder_left_front">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor1">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission> 
  <link
    name="low_front_left">
    <inertial>
      <origin
        xyz="0 0.0499999509898841 -7.0007208134687E-05"
        rpy="0 0 0" />
      <mass
         value="0.18" />
      <inertia
        ixx="0.0002842"
        ixy="3.93E-21"
        ixz="-1.11E-21"
        iyy="1.6E-05"
        iyz="3.75E-07"
        izz="0.00028" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/low_front_left.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/low_front_left.STL" />
      </geometry>
    </collision>
  </link>
  <gazebo reference="low_front_left">
    <kp>100000.0</kp>
    <kd>100000.0</kd>
    <mu1>1000000.0</mu1>
    <mu2>1000000.0</mu2>
    <material>Gazebo/Grey</material>
  </gazebo>
  <joint
    name="front_left_joint"
    type="revolute">
    <origin
      xyz="0 0.095 -0.038"
      rpy="0 0 0" />
    <parent
      link="upper_front_left" />
    <child
      link="low_front_left" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-15"
      upper="15"
      effort="10"
      velocity="1.0" />
  </joint>
   <transmission name="tran5">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="front_left_joint">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor5">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
  <link
    name="upper_front_right">
    <inertial>
      <origin
        xyz="-3.83232275913414E-08 0.0244785642847727 -0.0318067012629264"
        rpy="0 0 0" />
      <mass
        value="0.16224" />
      <inertia
        ixx="0.00015"
        ixy="2.269E-10"
        ixz="1.2008E-10"
        iyy="3.27E-05"
        iyz="2.459E-05"
        izz="0.000133" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/upper_front_right.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/upper_front_right.STL" />
      </geometry>
    </collision>
  </link>
   <gazebo reference="upper_front_right">
    <kp>100000.0</kp>
    <kd>100000.0</kd>
    <mu1>1000000.0</mu1>
    <mu2>1000000.0</mu2>
    <material>Gazebo/Grey</material>
  </gazebo>
  <joint
    name="shoulder_right_front"
    type="revolute">
    <origin
      xyz="-0.021 -0.225 0"
      rpy="-1.5708 0.52136 -1.5708" />
    <parent
      link="main_body" />
    <child
      link="upper_front_right" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-15"
      upper="15"
      effort="10"
      velocity="1.0" />
  </joint>
  <transmission name="tran2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="shoulder_right_front">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor2">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission> 
  <link
    name="low_front_right">
    <inertial>
      <origin
        xyz="6.93889390390723E-18 0.0499999509898841 -7.00072081346592E-05"
        rpy="0 0 0" />
      <mass
         value="0.18" />
      <inertia
        ixx="0.0002842"
        ixy="3.93E-21"
        ixz="-1.11E-21"
        iyy="1.6E-05"
        iyz="3.75E-07"
        izz="0.00028" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/low_front_right.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/low_front_right.STL" />
      </geometry>
    </collision>
  </link>
   <gazebo reference="low_front_right">
    <kp>100000.0</kp>
    <kd>100000.0</kd>
    <mu1>1000000.0</mu1>
    <mu2>1000000.0</mu2>
    <material>Gazebo/Grey</material>
  </gazebo>
  <joint
    name="front_right_joint"
    type="revolute">
    <origin
      xyz="0 0.095 -0.038"
      rpy="0 0 0" />
    <parent
      link="upper_front_right" />
    <child
      link="low_front_right" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-15"
      upper="15"
      effort="10"
      velocity="1.0" />
  </joint>
   <transmission name="tran6">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="front_right_joint">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor6">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission> 
  <link
    name="upper_back_left">
    <inertial>
      <origin
        xyz="-2.36636892858577E-08 0.0244785713376747 -0.0318067000775048"
        rpy="0 0 0" />
      <mass
       value="0.16224" />
      <inertia
        ixx="0.00015"
        ixy="2.269E-10"
        ixz="1.2008E-10"
        iyy="3.27E-05"
        iyz="2.459E-05"
        izz="0.000133" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/upper_back_left.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/upper_back_left.STL" />
      </geometry>
    </collision>
  </link>
   <gazebo reference="upper_back_left">
    <kp>100000.0</kp>
    <kd>100000.0</kd>
    <mu1>1000000.0</mu1>
    <mu2>1000000.0</mu2>
    <material>Gazebo/Grey</material>
  </gazebo>
  <joint
    name="shoulder_left_back"
    type="revolute">
    <origin
      xyz="0.141 -0.075 0"
      rpy="-1.5708 -0.52136 1.5708" />
    <parent
      link="main_body" />
    <child
      link="upper_back_left" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-15"
      upper="15"
      effort="10"
      velocity="1.0" />
  </joint>
   <transmission name="tran3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="shoulder_left_back">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor3">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission> 
  <link
    name="low_back_left">
    <inertial>
      <origin
        xyz="0 0.0499999509898841 -7.00072081346766E-05"
        rpy="0 0 0" />
      <mass
        value="0.18" />
      <inertia
        ixx="0.0002842"
        ixy="3.93E-21"
        ixz="-1.11E-21"
        iyy="1.6E-05"
        iyz="3.75E-07"
        izz="0.00028" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/low_back_left.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/low_back_left.STL" />
      </geometry>
    </collision>
  </link>
   <gazebo reference="low_back_left">
    <kp>100000.0</kp>
    <kd>100000.0</kd>
    <mu1>1000000.0</mu1>
    <mu2>1000000.0</mu2>
    <material>Gazebo/Grey</material>
  </gazebo>
  <joint
    name="back_left_joint"
    type="revolute">
    <origin
      xyz="0 0.095 -0.038"
      rpy="0 0 0" />
    <parent
      link="upper_back_left" />
    <child
      link="low_back_left" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-15"
      upper="15"
      effort="10"
      velocity="1.0" />
  </joint>
   <transmission name="tran7">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="back_left_joint">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor7">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission> 
  <link
    name="upper_back_right">
    <inertial>
      <origin
        xyz="-3.83232279382861E-08 0.0244785642847722 -0.0318067012629264"
        rpy="0 0 0" />
      <mass
         value="0.16224" />
      <inertia
        ixx="0.00015"
        ixy="2.269E-10"
        ixz="1.2008E-10"
        iyy="3.27E-05"
        iyz="2.459E-05"
        izz="0.000133" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/upper_back_right.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/upper_back_right.STL" />
      </geometry>
    </collision>
  </link>
   <gazebo reference="upper_back_right">
    <kp>100000.0</kp>
    <kd>100000.0</kd>
    <mu1>1000000.0</mu1>
    <mu2>1000000.0</mu2>
    <material>Gazebo/Grey</material>
  </gazebo>
  <joint
    name="shoulder_right_back"
    type="revolute">
    <origin
      xyz="-0.021 -0.075 0"
      rpy="-1.5708 0.52136 -1.5708" />
    <parent
      link="main_body" />
    <child
      link="upper_back_right" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-15"
      upper="15"
      effort="10"
      velocity="1.0" />
  </joint>
   <transmission name="tran4">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="shoulder_right_back">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor4">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
  <link
    name="low_back_right">
    <inertial>
      <origin
        xyz="6.93889390390723E-18 0.0499999509898841 -7.00072081346696E-05"
        rpy="0 0 0" />
      <mass
        value="0.18" />
      <inertia
        ixx="0.0002842"
        ixy="3.93E-21"
        ixz="-1.11E-21"
        iyy="1.6E-05"
        iyz="3.75E-07"
        izz="0.00028" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/low_back_right.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://my_rob_description/models/rob/low_back_right.STL" />
      </geometry>
    </collision>
  </link>
   <gazebo reference="low_back_right">
    <kp>100000.0</kp>
    <kd>100000.0</kd>
    <mu1>1000000.0</mu1>
    <mu2>1000000.0</mu2>
    <material>Gazebo/Grey</material>
  </gazebo>
  <joint
    name="back_right_joint"
    type="revolute">
    <origin
      xyz="0 0.095 -0.038"
      rpy="0 0 0" />
    <parent
      link="upper_back_right" />
    <child
      link="low_back_right" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-15"
      upper="15"
      effort="10"
      velocity="1.0" />
  </joint>
 <transmission name="tran8">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="back_right_joint">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="motor8">
            <hardwareInterface>EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission> 
</robot>